### ISSUE TITLE
Feature Request: Add information about file creator/last editor as extended file attributes

### ISSUE THREAD

**Issue by PMunch** [abraunegg/onedrive#2719]
### Is your feature request related to a problem? Please describe.

I'm using this to sync our company Sharepoint to my machine so I can more easily work with the files there. One thing I noticed which was missing was information about who last edited a file. This information can be very useful from time to time.

### Describe the solution you'd like

When files are downloaded or updated this program could store information about who last edited the file (and possibly other metadata) as [extended file attributes](https://man7.org/linux/man-pages/man7/xattr.7.html). 

### Describe alternatives you've considered

There's not really any other place to store information like this.

**Comment by abraunegg** [comment]
@PMunch 
This will require some investigations and will not make it into v2.5.0 - sorry.

You can however use `--modified-by` to see this data. Please read https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/application-config-options.md#cli-option---modified-by for details

**Comment by PMunch** [comment]
No worries, just figured it would be a nice addition. Especially in combination with a file browser to display it (unfortunately my [feature request](https://github.com/lxqt/pcmanfm-qt/issues/1905) to my current file browser doesn't seem to be going anywhere). Do you know if changing the extended attributes of a file would somehow mess with the sync? Otherwise it would be possible to write a script that checked each file and wrote the attributes. Not as clean or efficient as tying it into the syncing itself of course.

**Comment by abraunegg** [comment]
@PMunch

Please can you test the PR below that implements your feature request.

First install all the require platform dependencies to build the client on your respective platforms. Please read https://github.com/abraunegg/onedrive/blob/master/docs/install.md#building-from-source---high-level-requirements and then follow correctly for your platform.

> [!IMPORTANT]
> Currently there is an issue with installing the DMD or LDC compiler using the documentation instructions:
> `curl -fsS https://dlang.org/install.sh | bash -s dmd` or `curl -fsS https://dlang.org/install.sh | bash -s ldc`
>
> This is not something I can solve and I have emailed the appropriate folk and raised https://github.com/dlang/dmd/issues/20830
> 
> Please augment the installation of DMD or LDC with your distribution version if it is available

Once this is done, to clone the PR to resolve your issue, you can use a script like the following:
```
#!/bin/bash

PR=3113

rm -rf ./onedrive-pr${PR}
git clone https://github.com/abraunegg/onedrive.git onedrive-pr${PR}
cd onedrive-pr${PR}
git fetch origin pull/${PR}/head:pr${PR}
git checkout pr${PR}

# Configure and Build
./configure --enable-debug --enable-notifications; make clean; make;
./onedrive --version
```

This script will create a local folder called `onedrive-pr3113` with the PR version.

To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any other options needed>
```
To install the PR, you will need to perform `sudo make install` to install the PR version to your system.

When running the PR, your version should be: `onedrive v2.5.4-4-g9e39fcc` or greater.

To enable this new capability, add the following to your config file:
```
write_xattr_data = "true"
```

Any new files that are downloaded will have xattr data added:
```
security.selinux="unconfined_u:object_r:user_home_t:s0"
user.onedrive.createdBy="Account Display Name"
user.onedrive.lastModifiedBy="Account Display Name"
```

**Comment by abraunegg** [comment]
@PMunch 
Please can you respond in regards to your feature request

**Comment by PMunch** [comment]
Sorry, I've been sick for over ' week now so been pretty reduced in capacity. I'll see if I can't get this tested today!

**Comment by abraunegg** [comment]
@PMunch 
Any further update here?

**Comment by PMunch** [comment]
Sorry, I returned to work too early and got completely knocked out again. Still recovering. I'll let you know once I'm back and can test this out. Hopefully as early as Monday.

**Comment by abraunegg** [comment]
@PMunch 
Hope you are better - any update in any testing?

**Comment by abraunegg** [comment]
@PMunch 
> Of course it would be even better if it could add this information to already existing files as well.

If you perform a `--resync` .. that is exactly what it should do - add this data to existing files as well.

Please can you validate this.

I will mark this feature as implemented and post your validation on the `--resync` merge this into 'master'

**Comment by PMunch** [comment]
Tried it today, works perfectly!

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
